### PR TITLE
Fixes #4: Automate launcher catalog update

### LINKED ISSUE #4: ⚡️ for-all-boosters: update launcher catalog when release function is called
Automatically update and push a branch to launcher catalog when release is called.

### PR DESCRIPTION
**Pull request by metacosm**
Updated `catalog` function to perform launcher update automatically. Could be called from `release` as well, though this isn't implemented yet.

### CODE REVIEW

File: for-all-boosters.sh [hunk 1]
```diff
@@ -1268,6 +1392,15 @@ do
     fi
 done
 
+if [ -n "${postCmd}" ]; then
```
Comment (geoand) [inline for-all-boosters.sh]: Perhaps having an example in the README would be good to explain this feature

File: for-all-boosters.sh [hunk 2]
```diff
@@ -436,6 +436,45 @@ replace_template_runtime_version_of_booster() {
     fi
 }
 
+# based on https://stackoverflow.com/a/10583562 for the components result.
+# Note that to retrieve the components, the variable must be named 'components' on the calling site and needs to be 'eval'ed:
+# local components; components=eval $(parse_version 1.5.13-2-SNAPSHOT)
```
Comment (geoand) [inline for-all-boosters.sh]: Probably having a example for non "components" would be good as well (or at least a mention in the comments that the special handling needed for components does not apply to them)

File: for-all-boosters.sh [hunk 3]
```diff
@@ -436,6 +436,45 @@ replace_template_runtime_version_of_booster() {
...
+parse_version() {
```
Comment (geoand) [inline for-all-boosters.sh]: Can this function perhaps be used existing parts of the script as well?

Comment (metacosm) [reply for-all-boosters.sh]: You mean in other parts of the script? Yes, but that was the next step ;)

File: for-all-boosters.sh (hunk 1 above)
Comment (metacosm) [reply for-all-boosters.sh]: Yes.

File: for-all-boosters.sh (hunk 2 above)
Comment (metacosm) [reply for-all-boosters.sh]: Yes.

File: for-all-boosters.sh (hunk 3 above)
Comment (geoand) [reply for-all-boosters.sh]: Yes, exactly :)

File: for-all-boosters.sh (hunk 1 above)
Comment (geoand) [reply for-all-boosters.sh]: :+1:

File: for-all-boosters.sh (hunk 2 above)
Comment (geoand) [reply for-all-boosters.sh]: :+1:

**Comment by metacosm** [comment]
Updated.

File: for-all-boosters.sh [hunk 4]
```diff
@@ -718,9 +757,102 @@ run_smoke_tests() {
...
+}
+
+get_sb_version_file() {
+    echo "${WORK_DIR}/spring-version.txt"
+}
+
 catalog() {
-    # todo: update launcher catalog instead
-    echo "${BOOSTER}: ${BRANCH} => $(get_latest_tag)" >>"$CATALOG_FILE"
+    declare -Ar catalog_branch_mapping=( ["master"]="current-community" ["redhat"]="current-redhat" ["osio"]="current-osio" )
```
Comment (geoand) [inline for-all-boosters.sh]: At last, we can use "maps" :P

Comment (metacosm) [reply for-all-boosters.sh]: Yes, we might need to clean up the workarounds we used so far?

Comment (geoand) [reply for-all-boosters.sh]: It's definitely worth it!

Comment (geoand) [reply for-all-boosters.sh]: Perhaps a new ticket that we can work on when we have some time?

Comment (metacosm) [reply for-all-boosters.sh]: Could be part of #6

Comment (metacosm) [reply for-all-boosters.sh]: Added.

Comment (geoand) [reply for-all-boosters.sh]: Good idea

**Pull request merged**